src/pperm.cc

```diff
@@ -3550,16 +3550,17 @@ Obj OnSetsPPerm(Obj set, Obj f)
             }
         }
     }
-    if (reslen == 0) {
-        RetypeBagSM(res, T_PLIST_EMPTY);
-        return res;
-    }
+
     SET_LEN_PLIST(res, reslen);
     SHRINK_PLIST(res, reslen);
```

Comment:
Moving the code above downwards is a tiny optimization in case a huge input set is mapped to an empty list. I don't expect this to matter for general real world inputs